relevanceai/operations/dr/base.py

```diff
@@ -82,20 +90,107 @@ def fit_transform_documents(
             A list of documents with the original vector field and the new vector field.
 
         """
-        documents = list(self.filter_docs_for_fields([vector_field], documents))
+
+        documents = list(self.filter_docs_for_fields(vector_fields, documents))
         vectors = np.array(
-            self.get_field_across_documents(
-                vector_field, documents, missing_treatment="skip"
+            self.get_fields_across_documents(
+                vector_fields, documents, missing_treatment="skip"
             )
         )
+        vectors = vectors.reshape(-1, vectors.shape[1] * vectors.shape[2])  # hacky fix
         dr_vectors = self.fit_transform(vectors, dims=dims)
-        dr_vector_field_name = self.get_dr_vector_field_name(vector_field, alias)
-        self.set_field_across_documents(dr_vector_field_name, dr_vectors, documents)
+        del vectors  # free more memory, mainly for memory edgecases
```

Comment:
You need to garbage collect after deleting 
```
import gc
gc.collect()
```

Reply:
not 100% sure if this is necessary within a function but have added it.